src/2_storage.js

```diff
@@ -177,25 +179,15 @@ var cookies = function(perm) {
 			setCookie(prefix(key), value);
 		},
 		remove: function(key) {
-			var expires = '';
-			document.cookie = prefix(key) + '=; expires=' + expires + '; path=/';
+			removeCookie(key, false);
 		},
 		clear: function() {
-			var deleteCookie = function(cookie) {
-				document.cookie = cookie.substring(0, cookie.indexOf('=')) + '=;expires=-1;path=/';
-			};
 			var cookieArray = document.cookie.split(';');
 			for (var i = 0; i < cookieArray.length; i++) {
-				var cookie = cookieArray[i];
-				cookie = cookie.substring(1, cookie.length);
-				if (cookie.indexOf(BRANCH_KEY_PREFIX) !== -1) {
-					if (!perm && cookie.indexOf('branch_expiration_date=') === -1) {
-						deleteCookie(cookie);
-					}
-					else if (perm && cookie.indexOf('branch_expiration_date=') > 0) {
-						deleteCookie(cookie);
-					}
-				}
+				var cookie = cookieArray[i].trim();
+				var firstEqualSign = cookie.indexOf("=");
+				var cookieName = cookie.substring(0, firstEqualSign);
+				removeCookie(cookieName, true);
```

Comment:
seems like this would remove all cookies on the sites domain and not just the ones branch adds. I think we need to add back the additional checks. 

Also, is there a reason we are no longer using expiration dates on these?

Reply:
There is now a check to ensure that we don't remove all cookies on the site's domain:

https://github.com/BranchMetrics/web-branch-deep-linking/blob/9fb097ae05626a8bfe79668d2f3fdc66de54def2/src/2_storage.js#L200

There are test cases for this as well:

https://github.com/BranchMetrics/web-branch-deep-linking/blob/9fb097ae05626a8bfe79668d2f3fdc66de54def2/test/2_storage.js#L99-L101

https://github.com/BranchMetrics/web-branch-deep-linking/blob/9fb097ae05626a8bfe79668d2f3fdc66de54def2/test/2_storage.js#L180-L199

Reply:
Also, the expiration date area below is never hit since we don't use 'permcookie' as a storage option in the SDK:

https://github.com/BranchMetrics/web-branch-deep-linking/blob/master/src/2_storage.js#L142-L147

This means that we don't have to check for anything related to 'branch_expiration_date'